src/secret.js

```diff
@@ -53,19 +71,36 @@ export const checkSecretKey = async (options) => {
 		exit(1);
 	}
 
-	const { name } = options.context.repository;
-	const { siteId } = options.context.searchspring;
 	const keys = options.context.user.keys || {};
-	const secretKey = keys[siteId];
+	let siteId = options.context.searchspring.siteId;
+	let name = options.context.repository.name;
 
 	try {
-		try {
-			await new ConfigApi(secretKey, options.dev).validateSite(name, siteId);
-			console.log(chalk.green('Verification of siteId and secretKey complete.'));
-		} catch (err) {
-			console.log(chalk.red('Verification of siteId and secretKey failed.'));
-			console.log(chalk.red(err));
-			exit(1);
+		if (options.multipleSites.length) {
+			for (let i = 0; i < options.multipleSites.length; i++) {
+				const { siteId, name, secretKey } = options.multipleSites[i];
+				try {
+					await new ConfigApi(secretKey, options.dev).validateSite(name, siteId);
+					console.log(chalk.green(`Verification of siteId and secretKey complete for ${name}`));
+				} catch (err) {
+					console.log(chalk.red(`Verification of siteId and secretKey failed for ${name}`));
+					console.log(chalk.red(err));
+					exit(1);
+				}
+
+				// prevent rate limiting
+				await wait(1111);
+			}
+		} else {
```

Comment:
Could be more DRY here by refactoring repeated code into a function or something similar.